src/Compilers/Core/Portable/Serialization/ObjectWriter.cs

```diff
@@ -57,18 +57,8 @@ internal sealed partial class ObjectWriter : IDisposable
         /// and not the objects themselves because the ID is inferred from the order the object is
         /// written or read in.
         /// </summary>
-        private WriterReferenceMap _objectReferenceMap;
         private WriterReferenceMap _stringReferenceMap;
 
-        /// <summary>
-        /// Copy of the global binder data that maps from Types to the appropriate reading-function
-        /// for that type.  Types register functions directly with <see cref="ObjectBinder"/>, but
-        /// that means that <see cref="ObjectBinder"/> is both static and locked.  This gives us
-        /// local copy we can work with without needing to worry about anyone else mutating.
-        /// </summary>
-        private readonly ObjectBinderSnapshot _binderSnapshot;
-
-        private int _recursionDepth;
```

Comment:
yaay, there's no more recursion at all in serialization! :)

Reply:
Do we still need `ObjectReader._recursionDepth` then?
Also, do we need `MaxRecursionDepth`?

Reply:
we do not.  removing!